src/formats/LAMMPSAtom.cpp

```diff
@@ -0,0 +1,665 @@
+                shape = UnitCell::TRICLINIC;
+            }
+            line = file_.readline();
+            auto splitted = split(line, ' ');
+            if ((shape == UnitCell::ORTHORHOMBIC && splitted.size() != 2) ||
+                (shape == UnitCell::TRICLINIC && splitted.size() != 3)) {
+                size_t expected_dims = (shape == UnitCell::ORTHORHOMBIC) ? 2 : 3;
+                throw format_error(
+                    "incomplete box dimensions in LAMMPS format, expected {} but got {}",
+                    expected_dims, splitted.size());
+            }
+            double xlo = parse<double>(splitted[0]);
+            double xhi = parse<double>(splitted[1]);
+            matrix[0][0] = xhi - xlo;
+            origin[0] = xlo;
+            if (shape == UnitCell::TRICLINIC) {
+                matrix[0][1] = parse<double>(splitted[2]);
+            }
+
+            line = file_.readline();
+            splitted = split(line, ' ');
+            if ((shape == UnitCell::ORTHORHOMBIC && splitted.size() != 2) ||
+                (shape == UnitCell::TRICLINIC && splitted.size() != 3)) {
+                size_t expected_dims = (shape == UnitCell::ORTHORHOMBIC) ? 2 : 3;
+                throw format_error(
+                    "incomplete box dimensions in LAMMPS format, expected {} but got {}",
+                    expected_dims, splitted.size());
+            }
+            double ylo = parse<double>(splitted[0]);
+            double yhi = parse<double>(splitted[1]);
+            matrix[1][1] = yhi - ylo;
+            origin[1] = ylo;
+            if (shape == UnitCell::TRICLINIC) {
+                matrix[0][2] = parse<double>(splitted[2]);
+            }
+
+            line = file_.readline();
+            splitted = split(line, ' ');
+            if ((shape == UnitCell::ORTHORHOMBIC && splitted.size() != 2) ||
+                (shape == UnitCell::TRICLINIC && splitted.size() != 3)) {
+                size_t expected_dims = (shape == UnitCell::ORTHORHOMBIC) ? 2 : 3;
+                throw format_error(
+                    "incomplete box dimensions in LAMMPS format, expected {} but got {}",
+                    expected_dims, splitted.size());
+            }
+            double zlo = parse<double>(splitted[0]);
+            double zhi = parse<double>(splitted[1]);
+            matrix[2][2] = zhi - zlo;
+            origin[2] = zlo;
+            if (shape == UnitCell::TRICLINIC) {
+                matrix[1][2] = parse<double>(splitted[2]);
+            }
+
+            auto cell = UnitCell(matrix);
+            frame.set_cell(cell);
+            return origin;
+        } else {
+            throw format_error("missing 'BOX BOUNDS' item in LAMMPS format");
+        }
+    } catch (const Error& e) {
+        throw format_error("can not read box header in LAMMPS format: {}", e.what());
+    }
+}
+
+// posible per-atom attributes by dump command
+enum lammps_atom_attr_t {
+    UNKNOWN,
```

Comment:
Could you add small comments on all attributes telling what they are for someone not familiar with LAMMPS trajectory files?

Reply:
Added descriptions as provided by the [dump command (see `custom` style)](https://lammps.sandia.gov/doc/dump.html).